app/kuma-dp/pkg/dataplane/metrics/server.go

```diff
@@ -243,3 +363,52 @@ func (s *Hijacker) passRequestHeaders(into http.Header, from http.Header) {
 func (s *Hijacker) NeedLeaderElection() bool {
 	return false
 }
+
+const (
+	hdrContentType           = "Content-Type"
+	textType                 = "text/plain"
+	textVersion              = "0.0.4"
+	openmetricsVersion_1_0_0 = "1.0.0"
+	openmetricsVersion_0_0_1 = "0.0.1"
+	protoType                = `application/vnd.google.protobuf`
+	protoProtocol            = `io.prometheus.client.MetricFamily`
+)
+
+// responseFormat extracts the correct format from a HTTP response header.
+// If no matching format can be found FormatUnknown is returned.
+func responseFormat(h http.Header) expfmt.Format {
+	ct := h.Get(hdrContentType)
+
+	mediatype, params, err := mime.ParseMediaType(ct)
+	if err != nil {
+		return expfmt.FmtUnknown
+	}
+
+	version := params["version"]
+
+	switch mediatype {
+	case protoType:
+		p := params["proto"]
+		e := params["encoding"]
+		// only delimited encoding is supported by prometheus scraper
+		if p == protoProtocol && e == "delimited" {
+			return expfmt.FmtProtoDelim
+		}
+
+	// if mediatype is `text/plain`, return Prometheus text format
+	// without checking the version, as there are few exporters
+	// which don't set the version param in the content-type header. ex: Envoy
+	case textType:
+		return expfmt.FmtText
+
+	case expfmt.OpenMetricsType:
+		if version == openmetricsVersion_0_0_1 {
+			return expfmt.FmtOpenMetrics_0_0_1
+		}
+		if version == openmetricsVersion_1_0_0 {
+			return expfmt.FmtOpenMetrics_1_0_0
+		}
+	}
+
+	return expfmt.FmtUnknown
+}
```

Comment:
Ahh super! But even if the Prometheus PR is merged, it would return `expfmt.FmtUnknown` for Envoy, as Envoy does not include version parameter in its `Content-Type`.

But yes, by now I should have caught this already and used constants provided by `expfmt`.